### ISSUE TITLE
Error: Failed to download module 

### ISSUE THREAD

**Issue by tokiwong** [hashicorp/terraform#23824]
### Terraform Version
v0.12.18

### Crash Output
```
Could not download module "ec2_cluster" (main.tf:84) source code from
"https://api.github.com/repos/terraform-aws-modules/terraform-aws-ec2-instance/tarball/v2.0.0//*?archive=tar.gz":
Error opening a gzip reader for
/var/folders/62/xjfk0p_x78jdt6tx4rzl128c0000gp/T/getter331613727/archive: EOF.
```
### Expected Behavior
`terraform init` should have worked and downloaded the aws module from the corresponding [github repo](https://github.com/terraform-aws-modules/terraform-aws-security-group/releases/tag/v3.1.0)
### Actual Behavior
`terraform init` failed on `Error: failed to download module`
### Steps to Reproduce
1. Remove `.terraform/` folder
2. `terraform init`

**Comment by juanjbrown** [comment]
The URL is responding with a 404

**Comment by tokiwong** [comment]
Fixed by updating to `v0.12.19`
